src/dimensions.jl

```diff
@@ -208,45 +208,192 @@ function checklim(ctx::AbstractCompiler, a::FinchNode, b::FinchNode)
     end
 end
 
-@kwdef struct ParallelDimension{Ext,Device} <: AbstractExtent
+abstract type AbstractSchedule end
+
+abstract type AbstractVirtualSchedule end
+
+struct ZeroSchedule <: AbstractSchedule end
```

Comment:
We need to rename the schedules:

ZeroSchedule -> FinchStaticSchedule, static_schedule()
OneSchedule -> is broken plz remove
TwoSchedule -> FinchGreedySchedule, greedy_schedule()
ThreeSchedule -> FinchJuliaSchedule, julia_schedule()